thetis/timeintegrator.py

```diff
@@ -89,13 +89,13 @@ def update_solver(self):
         """Builds linear problems for each stage. These problems need to be
         re-created after each mesh update."""
         prob_k0 = LinearVariationalProblem(self.a_rk, self.L_RK0, self.K0)
-        self.solver_k0 = LinearVariationalSolver(prob_k0,
+        self.solver_k0 = LinearVariationalSolver(prob_k0, options_prefix=self.name + '_k0',
                                                  solver_parameters=self.solver_parameters)
         prob_k1 = LinearVariationalProblem(self.a_rk, self.L_RK1, self.K1)
-        self.solver_k1 = LinearVariationalSolver(prob_k1,
+        self.solver_k1 = LinearVariationalSolver(prob_k1, options_prefix=self.name + '_k1',
                                                  solver_parameters=self.solver_parameters)
         prob_k2 = LinearVariationalProblem(self.a_rk, self.L_RK2, self.K2)
-        self.solver_k2 = LinearVariationalSolver(prob_k2,
+        self.solver_k2 = LinearVariationalSolver(prob_k2, options_prefix=self.name + '_k2',
```

Comment:
a_rk is just a mass matrix I think, and can therefore be reused over the stages. If so we should use a LinearSolver which we can solve three times with the different right hand sides